test/unit/model/jdk-install-test.js

```diff
@@ -187,51 +182,46 @@ describe('JDK installer', function() {
             jdk.getMsiSearchScriptLocation(), jdk.getMsiSearchScriptData());
           expect(Util.executeFile).to.have.been.calledWith(
             'powershell', jdk.getMsiSearchScriptPowershellArgs(jdk.getMsiSearchScriptLocation()));
-          done();
         });
       });
     });
 
     describe('on macos', function() {
-      it('should not select jdk for installation if no java detected', function(done) {
+      it('should not select jdk for installation if no java detected', function() {
         sandbox.stub(Platform, 'getOS').returns('darwin');
         let jdk = new JdkInstall(installerDataSvc, 'url', 'jdk8.msi', 'jdk8', 'sha');
         mockDetectedJvm('');
         return jdk.detectExistingInstall(function() {
-          expect(jdk.selectedOption).to.be.equal('detected');
-          done();
+          expect(jdk.selectedOption).to.be.equal('No java detected');
```

Comment:
selectedOption can be install or detected, this test fails when checks to be placed inside then() callback.